### PR TITLE
Only check assertions that were meant to apply to the normal case of non-splat vector SREM expansion when we aren't hitting the special case.

### LINKED ISSUE #84830: assertion failing in multiple backends "We are expecting that A is always less than all-ones for SVT"
both the AArch64 and x86-64 backends assert out on this input:
```
define i32 @f(i1 %0) {
  %new0 = srem i1 %0, true
  %last = zext i1 %new0 to i32
  %2 = icmp ne i32 %last, 0
  %3 = select i1 %2, i32 0, i32 1
  ret i32 %3
}
```

### PR DESCRIPTION
**Pull request by resistor**
Fixes https://github.com/llvm/llvm-project/issues/84830
Introduced in https://github.com/llvm/llvm-project/pull/82706

### CODE REVIEW

File: llvm/test/CodeGen/AArch64/srem-vec-crash.ll [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+;RUN: llc -mtriple=aarch64-unknown-unknown < %s | FileCheck %s
+
+define i32 @f(i1 %0) {
+; CHECK-LABEL: f:
+; CHECK:       // %bb.0:
+; CHECK-NEXT:    mov w0, #1 // =0x1
+; CHECK-NEXT:    ret
+  %new0 = srem i1 %0, true
+  %last = zext i1 %new0 to i32
+  %2 = icmp ne i32 %last, 0
```
Comment (arsenm) [inline llvm/test/CodeGen/AArch64/srem-vec-crash.ll:11]: Use named values

File: llvm/test/CodeGen/AArch64/srem-vec-crash.ll [hunk 2]
```diff
@@ -0,0 +1,14 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 4
+;RUN: llc -mtriple=aarch64-unknown-unknown < %s | FileCheck %s
+
+define i32 @f(i1 %0) {
```
Comment (arsenm) [inline llvm/test/CodeGen/AArch64/srem-vec-crash.ll:4]: Use issue number in function / test name?

**Review by arsenm — APPROVED** [review]

File: llvm/test/CodeGen/AArch64/srem-vec-crash.ll (hunk 1 above)
Comment (resistor) [reply llvm/test/CodeGen/AArch64/srem-vec-crash.ll]: Done

File: llvm/test/CodeGen/AArch64/srem-vec-crash.ll (hunk 2 above)
Comment (resistor) [reply llvm/test/CodeGen/AArch64/srem-vec-crash.ll]: Done

**Pull request merged**